website/client-old/js/services/undoService.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+
+angular.module('habitrpg')
+.factory('Undo', ['$rootScope', '$http', 'Tasks', 'Common',
+  function undo($rootScope, $http, Tasks, Common) {
+    var undoApi = {};
+
+    undoApi.history = [];
+
+    undoApi.addAction = function (action, data) {
+      undoApi.history.push({action: action, data: data});
+    };
+
+    undoApi.undoAction = function () {
+      var pop = undoApi.history.pop();
+      if (!pop) return;
+
+      // @TODO: We need to add a pop interface for each action
+      if (pop.action === 'scoreTask') {
+        var direction = pop.data.direction;
+
+        if (direction === "up")  {

```

Comment:
yes, but I think we would get more questions if the undo didn't work exactly as an undo. But at the same time the only other option I see is not sending the request for 10 seconds and in that timeframe it's undoable, which would have the problem of not showing drops for 10 seconds so not ideal